CMakeLists.txt

```diff
@@ -4,6 +4,17 @@
 
 cmake_minimum_required(VERSION 3.12)
 
+if((DEFINED TI_EMSCRIPTENED) AND TI_EMSCRIPTENED)
```

Comment:
Maybe use cmake's like `option` command?

Reply:
Hmm. I added the option in #4093 already.

Reply:
But since `TaichiCore.cmake` is included after the `project(taichi)` call, that `option` isn't declared here. I didn't wanna redeclare the option either.

Reply:
If that's the case that option should be moved here probably. Since it's really a platform option not a taichi feature option

Reply:
that makes sense. I'll move it here.